proof/verifier_test.go

```diff
@@ -0,0 +1,613 @@
+package proof
+
+import (
+	"testing"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcec/v2/schnorr"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightninglabs/taproot-assets/asset"
+	"github.com/lightninglabs/taproot-assets/commitment"
+	"github.com/lightninglabs/taproot-assets/internal/test"
+	"github.com/stretchr/testify/require"
+)
+
+// createProofConfig is a configuration struct for creating proofs. Not all
+// options are available for all proof creation functions.
+type createProofConfig struct {
+	additionalPrevWitnesses []asset.Witness
+	addStxoProof            bool
+	addAdditionalStxoProofs bool
+	updateAnchorTx          bool
+}
+
+type createProofOpt func(*createProofConfig)
+
+func withAdditionalPrevWitnesses(
+	additionalPrevWitnesses []asset.Witness) createProofOpt {
+
+	return func(cfg *createProofConfig) {
+		cfg.additionalPrevWitnesses = additionalPrevWitnesses
+	}
+}
+
+func withAddStxoProof() createProofOpt {
+	return func(cfg *createProofConfig) {
+		cfg.addStxoProof = true
+	}
+}
+
+func withAddAdditionalStxoProofs() createProofOpt {
+	return func(cfg *createProofConfig) {
+		cfg.addAdditionalStxoProofs = true
+	}
+}
+
+func withUpdateAnchorTx() createProofOpt {
+	return func(cfg *createProofConfig) {
+		cfg.updateAnchorTx = true
+	}
+}
+
+// outputScript returns the output script for the given taproot commitment
+// and internal key.
+func outputScript(t *testing.T, c *commitment.TapCommitment,
+	i *btcec.PublicKey) []byte {
+
+	tapscriptRoot := c.TapscriptRoot(nil)
+	expectedKey, err := schnorr.ParsePubKey(schnorr.SerializePubKey(
+		txscript.ComputeTaprootOutputKey(i, tapscriptRoot[:]),
+	))
+	pkScript, err := txscript.PayToTaprootScript(expectedKey)
+	require.NoError(t, err)
+
+	return pkScript
+}
+
+// makeSpentAsset creates a random asset that spends the given witness.
+func makeSpentAsset(t *testing.T, witness asset.Witness) *asset.Asset {
```

Comment:
Cool, this is the exact function I think we can lift to live outside the test package, for use elsewhere.